### PR TITLE
add version centos7 for compatible

### LINKED ISSUE #2393: max_user_namespaces not enable warn when set up environment not start with user namespace
**Description**

when run buildah inside container, it shows warning of enable max_user_namespace
```
buildah bud -t buildah-compile .
Error during unshare(CLONE_NEWUSER): Invalid argument
User namespaces are not enabled in /proc/sys/user/max_user_namespaces.
ERRO error parsing PID "": strconv.Atoi: parsing "": invalid syntax 
ERRO (unable to determine exit status) 
```
It seems like I should enable user namespace using command like `echo 15000 > /proc/sys/user/max_user_namespaces`.
But what make me confused is that the [Dockerfile](https://github.com/containers/buildah/blob/master/contrib/buildahimage/stable/Dockerfile) of the image `quay.io/buildah/stable` shows it already set up env `_BUILDAH_STARTED_IN_USERNS=""` to stop start buildah with user namespace. 
1. if it's the problem of that I didn't enable user namespace, why the env is not work?
2. if it's not the problem of user namespace, how can I debug its root cause?

**Steps to reproduce the issue:**
``` bash
podman run -it --rm quay.io/buildah/stable bash
cd
echo -e "FROM quay.io/buildah/stable\nRUN dnf -y update;dnf -y install java-1.8.0-openjdk-devel nodejs maven;dnf -y clean all" > Dockerfile
buildah bud -t buildah-compile .
Error during unshare(CLONE_NEWUSER): Invalid argument
User namespaces are not enabled in /proc/sys/user/max_user_namespaces.
ERRO error parsing PID "": strconv.Atoi: parsing "": invalid syntax 
ERRO (unable to determine exit status)
```

**Describe the results you expected:**
buildah should work

**Output of `rpm -q buildah` or `apt list buildah`:**

```
# host
buildah-1.11.6-11.el7_8.x86_64

# container
buildah-1.14.8-1.fc32.x86_64
```

**Output of `buildah version`:**

```
# host
buildah version 1.11.6 (image-spec 1.0.1-dev, runtime-spec 1.0.1-dev)

# container
buildah version 1.14.8 (image-spec 1.0.1-dev, runtime-spec 1.0.1-dev)
```
**Output of `cat /etc/*release`:**

```
# host
CentOS Linux release 7.8.2003 (Core)

# container
Fedora release 32 (Thirty Two)
```

**Output of `uname -a`:**

```
# host
Linux localhost.localdomain 3.10.0-1062.4.1.el7.x86_64 #1 SMP Fri Oct 18 17:15:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

# container
Linux cdd19f290235 3.10.0-1062.4.1.el7.x86_64 #1 SMP Fri Oct 18 17:15:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```

**Output of `cat /etc/containers/storage.conf`:**

```
[storage]
driver = "overlay"
runroot = "/var/run/containers/storage"
graphroot = "/var/lib/containers/storage"
[storage.options]
additionalimagestores = [
"/var/lib/shared",
]
[storage.options.overlay]
mount_program = "/usr/bin/fuse-overlayfs"
mountopt = "nodev,fsync=0"
[storage.options.thinpool]
```

### PR DESCRIPTION
**Pull request by xiaotuanyu120**
Signed-off-by: xiaotuanyu120 <zhaopeiwu@outlook.com>

#### What type of PR is this?

> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test 
> /kind feature
> /kind flake

/kind other

#### What this PR does / why we need it:
add a new image branch centos7 / for compatible with centos7 and other distribution which kernel version is 3.10.0 and less

#### How to verify it
``` bash
# os version: centos7
# user: root

# git clone repo to local
mkdir storage
buildah -t buildah-centos7 bud buildah/contrib/buildahimage/centos7
podman run --rm -it --device /dev/fuse:rw -v ./storage:/var/lib/containers:Z buildah-centos7 bash
echo -e "From fedora\nRUN echo test" > Dockerfile
buildah -t test bud .
```

#### Which issue(s) this PR fixes:

Fixes #2393 

#### Does this PR introduce a user-facing change?

```release-note
For centos7, change image url from `quay.io/buildah/stable` to `quay.io/buildah/centos7`
```

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM
@TomSweeneyRedHat PTAL

File: contrib/buildahimage/centos7/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# stable/Dockerfile
```
Comment (TomSweeneyRedHat) [inline contrib/buildahimage/centos7/Dockerfile:1]:
```suggestion
# centos7/Dockerfile
```

File: contrib/buildahimage/centos7/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+# change fedora:latest to centos:7
+FROM centos:7.8.2003
+
+# Don't include container-selinux and remove
+# directories used by yum that are just taking
+# up space.
+# remove --exclude container-selinux
+RUN useradd build; yum -y update; yum -y reinstall shadow-utils; yum -y install buildah fuse-overlayfs; rm -rf /var/cache /var/log/dnf* /var/log/yum.*;
+
+ADD https://raw.githubusercontent.com/containers/buildah/master/contrib/buildahimage/centos7/containers.conf /etc/containers/
```
Comment (TomSweeneyRedHat) [inline contrib/buildahimage/centos7/Dockerfile:18]:
```suggestion
ADD https://raw.githubusercontent.com/containers/buildah/master/contrib/buildahimage/stable/containers.conf /etc/containers/
```

File: contrib/buildahimage/centos7/containers.conf [hunk 3]
```diff
@@ -0,0 +1,2 @@
+[engine]
```
Comment (TomSweeneyRedHat) [inline contrib/buildahimage/centos7/containers.conf:1]: Drop this file and just use the one from stable as noted in a comment above.

File: contrib/buildahimage/centos7/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# stable/Dockerfile
+#
+# Build a Buildah container image from the latest
+# stable version of Buildah on the Fedoras Updates System.
```
Comment (TomSweeneyRedHat) [inline contrib/buildahimage/centos7/Dockerfile:4]: Being Centos, it doesn't pull the image from the Fedoras Updates System does it?

File: contrib/buildahimage/centos7/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+# https://bodhi.fedoraproject.org/updates/?search=buildah
+# This image can be used to create a secured container
+# that runs safely with privileges within the container.
+#
+# change fedora:latest to centos:7
+FROM centos:7.8.2003
```
Comment (TomSweeneyRedHat) [inline contrib/buildahimage/centos7/Dockerfile:10]: could you use `centos:7` here?  Or `centos:latest`?

File: contrib/buildahimage/README.md [hunk 6]
```diff
@@ -44,3 +44,13 @@ buildah images
 
 exit
 ```
+
+## Compatible with old versions
+For compatible reason, some linux distribution with kernel 3.10.0 and less will not work when use the stable image which based on fedora:32. So this centos7 branch comes for fix this.
+
+Changes between it and stable:
+- change base image from fedora to centos:7.8.2003
```
Comment (TomSweeneyRedHat) [inline contrib/buildahimage/README.md:52]: if you change the image in the Dockerfile below, please also change this.

File: contrib/buildahimage/README.md [hunk 7]
```diff
@@ -44,3 +44,13 @@ buildah images
...
+- remove `--exclude container-selinux` when install buildah and fuse-overlayfs, for details pls check [here](https://bugzilla.redhat.com/show_bug.cgi?id=1806044)
```
Comment (TomSweeneyRedHat) [inline contrib/buildahimage/README.md:53]:
```suggestion
- remove `--exclude container-selinux` when installing buildah and fuse-overlayfs, for details pls check [here](https://bugzilla.redhat.com/show_bug.cgi?id=1806044)
```

File: contrib/buildahimage/README.md [hunk 8]
```diff
@@ -44,3 +44,13 @@ buildah images
...
+- update the sed logic of storage.conf(the final result is same, just change the method as the default file format is changed when fuse-overlayfs version is changed)
```
Comment (TomSweeneyRedHat) [inline contrib/buildahimage/README.md:54]:
```suggestion
- update the sed logic of storage.conf (the final result is the same, this change just changes the methodology)
```
I think that's what you're saying anyway.

File: contrib/buildahimage/README.md [hunk 9]
```diff
@@ -44,3 +44,13 @@ buildah images
 
 exit
 ```
+
+## Compatible with old versions
+For compatible reason, some linux distribution with kernel 3.10.0 and less will not work when use the stable image which based on fedora:32. So this centos7 branch comes for fix this.
```
Comment (TomSweeneyRedHat) [inline contrib/buildahimage/README.md:49]:
```suggestion
For compatibility reasons, some Linux distributions with a kernel version of 3.10.0 and less will not work when using the stable image that is based on fedora:32. This centos7 image can be used to work on those distributions.
```

**Comment by TomSweeneyRedHat** [comment]
Overall looks good to me, just a few mostly nit things here and there.  TYVM @xiaotuanyu120 for putting this together.

@rhatdan I'm assuming I should open a new repo on quay.io to autobuild this there once complete?  Holler if not.

**Comment by rhatdan** [comment]
@xiaotuanyu120 You need to squash your PRs and then force push.
git rebase -i origin
git push --force

**Comment by xiaotuanyu120** [comment]
> LGTM, TYVM @xiaotuanyu120 !

Thank you for correcting my grammar.

**Comment by xiaotuanyu120** [comment]
> @xiaotuanyu120 You need to squash your PRs and then force push.
> git rebase -i origin
> git push --force

Am I doing this right? (First time to do a PR, smile)

**Comment by TomSweeneyRedHat** [comment]
LGTM
@rhatdan, can you double check the squash and we can merge.

**Comment by philippe-granet** [comment]
@TomSweeneyRedHat I try to use buildah on a Centos 7.6. Is it possible to publish this image on quay.io please ?

**Pull request merged**